xbmc/pvr/windows/GUIWindowPVRGuide.cpp

```diff
@@ -616,8 +616,25 @@ bool CGUIWindowPVRGuideBase::RefreshTimelineItems()
 
       std::unique_ptr<CFileItemList> timeline(new CFileItemList);
 
-      // can be very expensive. never call with lock acquired.
-      group->GetEPGAll(*timeline, true);
+      if (m_vecItems->IsEmpty() && !m_newTimeline)
+      {
+        // very first open of the window. come up with some data very fast...
+        const std::vector<PVRChannelGroupMember> groupMembers = group->GetMembers();
+        for (const auto& groupMember : groupMembers)
+        {
+          // fake a channel without epg
+          const std::shared_ptr<CPVREpgInfoTag> gapTag = std::make_shared<CPVREpgInfoTag>(groupMember.channel);
+          timeline->Add(std::make_shared<CFileItem>(gapTag));
```

Comment:
If this adding of "gap tags" is used elsewhere too it might make sense to add an AddGapTag(channel) method

Reply:
I will put this on my list for another day. Thanks.